src/state/zustand/swapperStore/amountSelectors.ts

```diff
@@ -411,10 +290,10 @@ export const selectBuyAmountAfterFeesFiat = createSelector(
 
 export const selectQuoteBuyAmountCryptoPrecision = createSelector(
   selectTradeOrQuoteBuyAmountCryptoBaseUnit,
-  (state: SwapperState) => state.buyAsset?.precision,
-  (tradeOrQuoteBuyAmountCryptoBaseUnit, buyAssetPrecision): string | undefined => {
-    if (!tradeOrQuoteBuyAmountCryptoBaseUnit || !buyAssetPrecision) return undefined
-    return fromBaseUnit(tradeOrQuoteBuyAmountCryptoBaseUnit, buyAssetPrecision)
+  selectBuyAsset,
+  (tradeOrQuoteBuyAmountCryptoBaseUnit, buyAsset): string | undefined => {
+    if (!tradeOrQuoteBuyAmountCryptoBaseUnit) return undefined
```

Comment:
Early return if no `buyAsset?.precision` still? `(parameter) buyAsset: Asset` is the types lying to us, the buy `Asset` might very well be undefined

Reply:
`state.buyAsset` is always defined since https://github.com/shapeshift/web/commit/10b64825a4f924aabf2847f7c4c5902a59102b4c#diff-1e1ff1d553c63056014c0b8490eb4c4898094c160402247649ca4211b1cc1d1c , but when that change went in the selectors weren't rewritten to take this into account